api-server-rest/src/cdh.rs

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2023 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use crate::router::ApiHandler;
+use crate::ttrpc_proto::confidential_data_hub::GetResourceRequest;
+use crate::ttrpc_proto::confidential_data_hub_ttrpc::GetResourceServiceClient;
+use anyhow::*;
+use async_trait::async_trait;
+use hyper::{Body, Method, Request, Response};
+use std::net::SocketAddr;
+
+use crate::utils::split_nth_slash;
+use crate::TTRPC_TIMEOUT;
+
+/// ROOT path for Confidential Data Hub API
+pub const CDH_ROOT: &str = "/cdh";
+
+/// URL for querying CDH get resource API
+pub const CDH_RESOURCE_URL: &str = "/resource";
+
+const KBS_PREFIX: &str = "kbs://";
+
+pub struct CDHClient {
+    client: GetResourceServiceClient,
+    accepted_method: Vec<Method>,
+}
+
+#[async_trait]
+impl ApiHandler for CDHClient {
+    async fn handle_request(
+        &self,
+        remote_addr: SocketAddr,
+        url_path: &str,
+        req: Request<Body>,
+    ) -> Result<Response<Body>> {
+        if !remote_addr.ip().is_loopback() {
```

Comment:
What is the purpose of this check?

Reply:
Current `api-server-rest` listen on local host which only can be accessed by internal signed container workload, we also support export external server if needed, add this check in current `cdh` and `aa` endpoint is to avoid the info leak if we switch to listen on public IP address.